classes/Hook.php

```diff
@@ -995,4 +1004,237 @@ private static function getHookRegistry()
 
         return null;
     }
+
+    /**
+     * Retrieves all modules registered to any hook, indexed by hok name.
+     *
+     * Each registration looks like this:
+     *
+     * ```
+     *     [
+     *         'id_hook' => $hookId,
+     *         'module' => $moduleName,
+     *         'id_module' => $moduleId
+     *     ]
+     * ```
+     *
+     * @param Context $context
+     * @param string|null $hookName Hook name (to be used when the hook registration is dynamic and context sensitive)
+     *
+     * @return array[][]
+     *
+     * @throws PrestaShopDatabaseException
+     */
+    private static function getAllHookRegistrations(Context $context, ?string $hookName): array
+    {
+        $shop = $context->shop;
+        $customer = $context->customer;
+
+        $cache_id = self::MODULE_LIST_BY_HOOK_KEY
+            . ($shop instanceof Shop && isset($shop->id) ? '_' . $shop->id : '')
+            . ($customer instanceof Customer ? '_' . $customer->id : '');
```

Comment:
```suggestion
            . (isset($shop->id) ? '_' . $shop->id : '')
            . (isset($customer->id) ? '_' . $customer->id : '');
```
Instanceof checks are useless because of isset